### ISSUE TITLE
Docs feature request: Local Change Detection Guide

### ISSUE THREAD

**Issue by e-oz** [angular/angular#52953]
### Describe the problem that you experienced

Please add a guide (or any other documentation) for the Local Change Detection.

What I know: 
* the component should use OnPush CD and the template should use signals.

What is not known:
* if the template will have at least one function call (except the signals unwrapping), will it switch this component to a regular OnPush CD?
* if the template will have at least one usage of async pipe - will it break local change detection?
* Will impure pipes work with Local Change Detection?

---
_There is no template for docs feature requests, so I'm using this one._

**Comment by JeanMeche** [comment]
To answser the question, if anyone wants to make the change:

* A signal (consumed in the template) update marks the component as dirty and the parents as "traversal". 
* Anything else that calls `markForCheck()` (like the `AsyncPipe)` will mark the parents as dirty. 
* In both cases CD only happens on a `appRef.tick()`.

**Comment by stasberkov** [comment]
@JeanMeche Please answer my question regarding change detection in OnPush component and signal.

Documentation https://angular.dev/guide/signals#reading-signals-in-onpush-components says that if I read signal from template, angular tracks signal as a dependency. **But what will happen if I read signal from a component?** Will angular also track that signal as a dependency?

E.g. In template a component has logic like `@if (isVisible("preference-1")) { ... }` for conditionally showing/hiding other controls. In component it has function `public isVisible(pref: string): boolean { return this.prefService.isVisibleSignal(); }`.
Will in this case angular track `this.prefService.isVisibleSignal` signal as pendency?
Will it mark the component for change detection that called `this.prefService.isVisibleSignal` when the signal changes?

I created small example that demonstrated this behavior https://stackblitz.com/edit/angular-onpush-signal-call-in-component
Is it working as expected? Is it a bug? Can you please also add clarification to https://angular.dev/guide/signals ?

**Comment by JeanMeche** [comment]
Yes it will, because the function is invoked from a reactive context : the template.

**Comment by JeanMeche** [comment]
After a second though, I'd argue that this is pretty much an implementation detail that could be narrowed down to : If you use signal and OnPush, Angular will ensure to optimize CD.
